### ISSUE TITLE
Surprising behavior on u.if when target object does not have the key being updated

### ISSUE THREAD

**Issue by mark-rushakoff** [substantial/updeep#35]
When the predicate is truthy, everything works fine:

```
u({x: u.if(true, 1)}, {}) // {x: 1}
```

When the predicate is falsy, the behavior is surprising to me:

```
u({x: u.if(false, 1)}, {}) // Expected {} but actual was {x: {}}
```

**Comment by aaronjensen** [comment]
Hmm, yeah I can see how that is surprising. We aggressively default things, I could see this being an exception to that because you're not specifying an else. I'm not sure of the best way to do that, however, it'd have larger effects... specifically it would require us to not default if there are no updates at all. I'll think on this, if you want to pull request to further the discussion it'd be great. Thanks!

**Comment by mark-rushakoff** [comment]
I should have included this in the original post, but our workaround was essentially:

```
function updateX(x) {
  return someCondition ? someValue : x;
}

u({x: updateX}, target)
``` 

Obviously longer than `u.if` but it gets the job done.

I don't know enough about how other people are using updeep to say which approach is a better fit in general.

**Comment by aaronjensen** [comment]
Ahhh, ok that's not good. Fixed in 0.9.0. Thanks!
